File: 1742. Maximum Number of Balls in a Box/solution.cpp
class Solution {
public:
  int countBalls(int lowLimit, int highLimit) {
    unordered_map<int, int> dictionary;

    for (int num = lowLimit; num < highLimit + 1; num++){
      int total = 0;
      string stringNum = to_string(num);
      for (int charNum = 0; charNum < stringNum.size(); charNum++){
        total += (int) stringNum[charNum] - 48;
      }
      dictionary[total]++;
    }

    int highestCount = 0;
    for (auto num : dictionary){
      if (num.second > highestCount){
        highestCount = num.second;
      }
    }

    return highestCount;
  }
};

File: 1742. Maximum Number of Balls in a Box/solution.py
class Solution(object):
  def countBalls(self, lowLimit, highLimit):
    """
    :type lowLimit: int
    :type highLimit: int
    :rtype: int
    """
    dictionary = {}

    for num in range(lowLimit, highLimit + 1):
      total = 0
      for value in str(num):
        total += int(value)
        
      if total not in dictionary:
        dictionary[total] = 1
      else:
        dictionary[total] += 1

    highestCount = 0
    for key in dictionary:
      if dictionary[key] > highestCount:
        highestCount = dictionary[key]

    return highestCount